### DISCUSSION TITLE
Attribute error in cv2.face

### DISCUSSION THREAD

**Discussion by Naman065Malik** [datasciencecompleteguide/Machine-Learning#10] [Q&A] [answered]
I am trying to make a face lock system for my virtual assistant (jarvis) but in this procedure there is a attribute error in cv2.face module. I already try pip install opencv-contrib-python. surprisingly it worked in my trainer generator I don't know why is it not working in my face lock system.  Please Help!!!!
![Screenshot 2021-06-26 231609](https://user-images.githubusercontent.com/86320160/123521508-a71c6980-d6d4-11eb-8d6f-ccd6f288f4b8.png)
![Screenshot 2021-06-26 231644](https://user-images.githubusercontent.com/86320160/123521510-aedc0e00-d6d4-11eb-9f11-52ab2168b3b9.png)

**Comment by datasciencecompleteguide** [comment]
![naman_lbphfacerecognizer](https://user-images.githubusercontent.com/69833825/123521990-d4b6e200-d6d7-11eb-8bea-a54d43cd786f.PNG)

try installing both 
opencv-contrib-python
and 
opencv-contrib-python-headless

I think it will work for u as well!!

**Comment by datasciencecompleteguide** [comment, marked as answer]
> 
> 
> I am trying to make a face lock system for my virtual assistant (jarvis) but in this procedure there is a attribute error in cv2.face module. I already try pip install opencv-contrib-python. surprisingly it worked in my trainer generator I don't know why is it not working in my face lock system. Please Help!!!!
> ![Screenshot 2021-06-26 231609](https://user-images.githubusercontent.com/86320160/123521508-a71c6980-d6d4-11eb-8d6f-ccd6f288f4b8.png)
> ![Screenshot 2021-06-26 231644](https://user-images.githubusercontent.com/86320160/123521510-aedc0e00-d6d4-11eb-9f11-52ab2168b3b9.png)

check the spelling in the line 3
its 
cv2.face.LBPHFaceRecognizer_create()
not
cv2.face.LBPHFaceRRecognizer_create()

**Reply by Naman065Malik** [reply]
bro I tried both of your suggestions but nothing work. In fact i already installed all related packages according to me but still it showing error.
![Screenshot 2021-06-28 210828](https://user-images.githubusercontent.com/86320160/123664651-2638ab80-d855-11eb-8bd6-a14cffc3f8b0.png)

**Reply by datasciencecompleteguide** [reply]
check the spelling of **_LBPHFaceRRecognizer_create()_**
its
cv2.face.LBPHFaceRecognizer_create()
not
cv2.face.LBPHFaceRRecognizer_create()

**Reply by Naman065Malik** [reply]
Bro i tried it also but it is  not working. when you last time told me to install packages then i saw  this mistake and rectified it .
but it is still not working.
![Screenshot 2021-06-30 225806](https://user-images.githubusercontent.com/86320160/124005540-d396f500-d9f6-11eb-8a09-7e1995cbce9a.png)
![Screenshot 2021-06-30 225834](https://user-images.githubusercontent.com/86320160/124005539-d396f500-d9f6-11eb-9626-f65f1162b30d.png)

**Reply by datasciencecompleteguide** [reply]
Do uninstall all:
opencv-python
opencv-contrib-python
opencv-contrib-python-headless

and then again install them.
